extensions-core/google-extensions/src/main/java/org/apache/druid/storage/google/GoogleStorage.java

```diff
@@ -142,15 +142,12 @@ public GoogleStorageObjectMetadata getMetadata(
     );
   }
 
-  public void delete(final String bucket, final String path) throws IOException
+  public void delete(final String bucket, final String path)
   {
-    if (!storage.get().delete(bucket, path)) {
-      throw new IOE(
-          "Failed deleting google cloud storage object [bucket: %s path: %s]",
-          bucket,
-          path
-      );
-    }
+    // Though currently not documented for the GCS delete api, a false response is indicative of file not found.
+    // All other errors appear as StorageException which is a runtime exceptions. Refer to
+    // https://github.com/googleapis/java-storage/blob/0b5f11af941032e6a55b12d243acf128a6464400/google-cloud-storage/src/main/java/com/google/cloud/storage/spi/v1/HttpStorageRpc.java#L685
+    storage.get().delete(bucket, path);
```

Comment:
Can we add a warn logging if the result is false?

Reply:
Done